src/Apps/Order/Routes/Shipping2/Components/FulfillmentDetailsForm.tsx

```diff
@@ -175,15 +182,19 @@ const FulfillmentDetailsFormLayout = (
   const tabbableIf = (activeForm: AddressFormMode): 0 | -1 =>
     addressFormMode === activeForm ? 0 : -1
 
-  const handleSelectSavedAddress = (address: ShippingAddressFormValues) => {
-    setValues({
+  const handleSelectSavedAddress = async (
+    address: ShippingAddressFormValues
+  ) => {
+    await setValues({
```

Comment:
I had to upgrade the formik version to give this the [correct Promise return type](https://github.com/jaredpalmer/formik/pull/3871) (so that `submitForm()` will hopefully work as expected)